Move to Zephyr 3.6.0

Deedone opened the pull request.

xakep-amatop committed:
boards: arm64: add support of Renesas RCar Spider board

Add support of r8a779f SoC and appropriate dts file
with minimal set of nodes.

Add support of 'rcar_spider_ca55' board: minimal dts
and configuration.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>

xakep-amatop committed:
tests: add h3ulcb, salvator xs m3 and spider configs to perf disk tests

Add MMC related cfgs/overlays of arm64 gen3/gen4 rcar boards to perf and
susys/sd/mmc tests.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>

xakep-amatop committed:
dts: arm64: add gpio5 and sd0 nodes to h3ulcb and salvator xs

Add gpio5 and sd0 nodes to h3ulcb and salvator xs which are needed
for working with SD cards. The GPIO node is needed for switching
voltage on SD card through gpio regulator driver.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

xakep-amatop committed:
samples: littlefs: add support for H3ULCB, Salvator XS M3 and Spider

Add support of R-Car H3ULCB, Salvator XS M3 and Spider boards to
littlefs sample.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>

oleksiimoisieiev committed:
drivers: xen: add Xen extended regions driver

Introducing driver for the extended regions, provided by XEN to
Zephyr, running as Xen Domain-0.
Extended regions are the unmapped space in Stage 2 and is not
bound to any guest and can be used by Zephyr for the foreign mapping.
It configures with regs property in hypervisor node, compatible to
xen,xen. For Example the following device-tree overlay:

&hypervisor {
	    reg = < 0x0 0x40200000 0x0 0x1000000
	    	    0x0 0x40400000 0x0 0x1FC00000
		    0x0 0x80000000 0x1 0x40000000>;
};

Will allocate two extended regions:
- 0x40400000 with size 0x1fc00000
- 0x80000000 with size 0x140000000

The first region: 0x40200000 size 0x1000000 is for the grant_tables
and will be ignored.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

oleksiimoisieiev committed:
devicetree: introduce new helper macro DT_FOREACH_REG

This macro helps iterating regs in the device-tree node.
The behavior is the same as for the ranges property and helps to the
allocations for each provided registry.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

oleksiimoisieiev committed:
drivers: xen: update gnttab driver to use xen extended regions

In gnttab driver gnttab_get_page and gnttab_put_page functions can use
extended regions for memory allocation. Added XEN_REGIONS support and
left original behavior when driver is disabled.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

oleksiimoisieiev committed:
boards: snippets: add extended region description to dom0 snippet

Add the most common region to the dom0 snippet that should be provided
by XEN. This exact region was selected because it is common for
renesas_rcar gen3 and renesas_rcar gen4 boards.
If Xen provide different regions - this value should be overridden in
the application.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

firscity committed:
drivers: xen: return hypercall results from notify_evtchn

Xen event channel notification may fail during hypervisor handling,
so it will be good for user to have a possibility to check results.

Previous implementation ignored hypervisor return code, now it will be
passed to caller.

Signed-off-by: Dmytro Firsov <dmytro_firsov@epam.com>

firscity committed:
drivers: xen: export missing evtchn API functions

Previously some functions, that are a part of Xen event channel API
were missing in events.h header (clear_event_channel() was also marked
as 'static').

Add missing calls to events.h and add documentation description for
to let users add it to external drivers.

Signed-off-by: Dmytro Firsov <dmytro_firsov@epam.com>

dsemenets committed:
xen: Add xen version hypercall

This hypercall used to obtain version of xen hypervisor

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Mykola Kvach <mykola_kvach@epam.com>

dsemenets committed:
xen: Add sysctl hypercall

This hypercall can be used get some information about physical machine
and running guests:

- sysctl hypercall "xen_sysctl_getphysinfo" allows read information about
physical machine: number CPUs, memory sizes, hardware capabilities, etc.

- sysctl hypercall "xen_sysctl_getdomaininfolist" returns array of domain
info structures that provide information about particular domain(s).

- sysctl hypercall "get_cpu_info" provides information about physical
CPU(s).

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Mykola Kvach <mykola_kvach@epam.com>

dsemenets committed:
xen: Added getvcpuinfo function in domctl hypercall

domctl hypercall "getvcpu" provides information about domain's
virtual CPU(s).

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Mykola Kvach <mykola_kvach@epam.com>

oleksiimoisieiev committed:
drivers: tee: Add Tee driver generic API

Implementation of the generic API Trusted Execution Environment driver.
This implements all basic syscalls, needed by tee driver, basic
defines and helpers to work with shared memory.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Mykola Kvach <mykola_kvach@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

oleksiimoisieiev committed:
drivers: tee: optee: Introducing OP-TEE driver implementation

This includes the basic TEE driver api implementation for OP-TEE.
It provides an interface to the OP-TEE TrustZone from the Normal space
to start sessions and request functions of the TA.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Mykola Kvach <mykola_kvach@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

oleksiimoisieiev committed:
drivers: tee: optee: Moving optee smc interface from optee client

Moved helper headers, needed to work with OP-TEE, which defines basic
defines, used during communication with Trust-Zone

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Reviewed-by: Sergiy Kibrik <sergiy_kibrik@epam.com>
Acked-by: Mykola Kvach <mykola_kvach@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

oleksiimoisieiev committed:
dts: bindings: Introduce device-tree bindings for OP-TEE

This introduces dts mapping description for the optee node.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Reviewed-by: Sergiy Kibrik <sergiy_kibrik@epam.com>
Acked-by: Mykola Kvach <mykola_kvach@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

oleksiimoisieiev committed:
tests: drivers: optee: Introduce unit tests for the OP-TEE driver

This should test OP-TEE driver basic functionality without
communication with real OPTEE OS.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Mykola Kvach <mykola_kvach@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

xakep-amatop committed:
drivers: tee: optee: add support of shared memory cmds

Add support of shared memory register/unregister to optee driver.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>
Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Mykola Kvach <mykola_kvach@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

oleksiimoisieiev committed:
tests: drivers: optee: Unit tests for shm functionality

Set SRAM_SIZE parameter to avoid assertions and setup smc callbacks
needed for register/unregister shm testing.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Mykola Kvach <mykola_kvach@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

oleksiimoisieiev committed:
drivers: tee: optee: Moving optee_rpc_cmd header

Moving header with OPTEE_RPC* defines needed to handle requests from
OP-TEE.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Mykola Kvach <mykola_kvach@epam.com>
Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

oleksiimoisieiev committed:
drivers: tee: optee: Introducing OPTEE_RPC_CMD function handles

This includes the implementation of the basic RPC_CMD which are
received from OP-TEE TA service.
Memory allocation requests, gettime and suspend were implemented in
this commit.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Mykola Kvach <mykola_kvach@epam.com>
Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

oleksiimoisieiev committed:
tests: drivers: optee: Unit tests for OPTEE_RPC_CMD_* handlers

This tests the behaviour of the rpc handlers, based on the Zephyr unit
tests architecture.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

xakep-amatop committed:
drivers: xen: regions: introduce new functionality for map/unmap ext mem

Introduce a new functionality for mapping and unmapping extended region
memory. It is needed to map ext mem after stage 2 mapping, i.e. hypervisor.

If any access to memory is performed before stage 2 mapping, we get
a MMU translation error.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>
Reviewed-by: Volodymyr Babchuk <volodymyr_babchuck@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

oleksiimoisieiev committed:
drivers: tee: optee: Introducing OPTEE_RPC_NOTIFICATION handlers

OPTEE_RPC_NOTIFICATION works as the synchronization for the Secure
World. It allows OP-TEE to request WAIT command, which makes Normal
World to sleep until different thread request for OP-TEE will cause
SEND commend to be sent which will unlock waiting thread.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Mykola Kvach <mykola_kvach@epam.com>
Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

oleksiimoisieiev committed:
tests: drivers: optee: Unit tests for OPTEE_CMD_NOTIFICATION

Tests lock/unlock mechanism for OP-TEE. Process lock/unlock from
different threads and tests send bit when SEND was received before
WAIT command.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

Deedone committed:
tests: drivers: optee: Unit tests for supplicant comms

Tests simple supplicant communication success path.

Signed-off-by: Mykyta Poturai <mykyta_poturai@epam.com>
Acked-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchyk@epam.com>

Deedone committed:
drivers: tee: optee: Implement SHM_TYPE_APPL allocation

Form and pass the allocation requests to the supplicant when
SHM_TYPE_APPL is requested by the OPTEE. Register allocated region in
the OPTEE afterwards.

Signed-off-by: Mykyta Poturai <mykyta_poturai@epam.com>
Acked-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Mykola Kvach <mykola_kvach@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchyk@epam.com>

Deedone committed:
tests: drivers: optee: Unit tests for supplicant alloc

Basic tests for supplicant shared memory allocation.

Signed-off-by: Mykyta Poturai <mykyta_poturai@epam.com>
Acked-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchyk@epam.com>

Deedone committed:
drivers: optee: Implement rev and caps negotiation

Implement initial uid checking, revision checking and capabilities
exchange. Capabilities are stored in the driver data for later use.

Signed-off-by: Mykyta Poturai <mykyta_poturai@epam.com>
Acked-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Mykola Kvach <mykola_kvach@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

Deedone committed:
drivers: optee: Fix typo in attr calculation

Change INOUT to INPUT to align the calculation with the rest of the
logic.

Signed-off-by: Mykyta Poturai <mykyta_poturai@epam.com>
Acked-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Mykola Kvach <mykola_kvach@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchyk@epam.com>

Deedone committed:
drivers: optee: Change signatures for supplicant send/recv

Change tee_suppl_recv signature to allow for returning of func and
num_params variables.

Change tee_supl_send signature to allow supplicant to provide the return
code.

This commit should be squashed with
f953fc5a466dfa54a0be2a234e6e7458d2515c8a
drivers: tee: Add Tee driver generic API
before upstreaming.

Signed-off-by: Mykyta Poturai <mykyta_poturai@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchyk@epam.com>

Deedone committed:
drivers: tee: optee: Implement supplicant communication

Implement suppl_send/suppl_receive syscalls and optee_call_supp function.

Requests passed to optee_call_supp are forwarded to the supplicant, the
calling thread gets blocked while the request is processing.

Signed-off-by: Mykyta Poturai <mykyta_poturai@epam.com>
Acked-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Mykola Kvach <mykola_kvach@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchyk@epam.com>

oleksiimoisieiev committed:
drivers: tee: optee: Introduce TMPMEM parameter handling

TMPMEM parameters are used for TA_LOAD call when OP-TEE requests
TA from REE storage.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>
Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

xakep-amatop committed:
drivers: sdhc: add support of Renesas MMC driver

Add basic functionality of Renesas SD/MMC driver. It can be used
for both gen3 and gen4 R-car boards, but tested only with H3ULCB,
Salvator XS M3 and Spider boards. This driver working with SDHC
subsystem.

The driver supports regularal reading/writing throught SD/MMC
controller buffer, DMA mode w/o interrupts and timing tuning.

Add gpio5 and sd0 nodes to h3ulcb and salvator xs which are needed
for working with SD cards. The GPIO node is needed for switching
voltage on SD card through gpio regulator driver.

Notes:
    * the driver doesn't support SPI mode;
    * SCC tuning and DMA mode based on IRQs are enabled by default;
    * an address of a data buffer has to be aligned to 128 bytes if it
      is not, driver will use non-DMA mode automatically;
    * Renesas MMC DMAC doesn't support 64-bit DMA addresses, so for
      case when we have 64-bit xref data address we use non-DMA mode;
    * SD/MMC controller supports block size between 512 and 1 with
      a lot of restrictions, more details you can find in code;
    * support of HS400 mode isn't implemented inside driver.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>

xakep-amatop committed:
boards: arm64: rcar: add board-specific configuration for MMC node

Add pin control group for UHS modes for H3ULCB, Salvator XS M3
boards to appropriate dts files. Both 'uhs' and 'default' pins
states have the same properties for eMMC, e.g. 1.8V.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>

xakep-amatop committed:
samples: shell: fs: add rcar_h3ulcb_ca57 support to shell/fs sample

Add config and overlay file for rcar_h3ulcb_ca57 board support to
subsys/shell/fs sample.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>
Acked-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

oleksiimoisieiev committed:
tests: drivers: optee: Fix API UID mismatch on test start

Test suite should set optee UID or the following error message should
be displayed:
[00:00:00.000,000] <err> optee: OPTEE API UID mismatch

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

oleksiimoisieiev committed:
tests: drivers: optee: Handle exchange capabilities call

Capabilities should be set correctly or optee will show error message:
[00:00:00.000,000] <err> optee: OPTEE does not support dynamic shared
memory

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

oleksiimoisieiev committed:
tests: drivers: tee: Fix normal_call behaviour in tests

During SMC_RPC_FUNC_ALLOC call optee driver puts new shm object to a4
and a5 variables and then expects this object to be in a1 and a2 from
OP-TEE OS. Fixed behaviour so normal_call callback will act the same
way as OP-TEE OS

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

oleksiimoisieiev committed:
drivers: tee: optee: Handle SHM_CACHE capability

OP-TEE can cache shm handles and reuse them across functions. This
will reduce allocations on Zephyr side. Before enabling shm_caching
driver does cleanup of the handles, registered in the OP-TEE from the
previous work. This is needed because Zephyr doesn't have destroy routine.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

oleksiimoisieiev committed:
drivers: tee: optee: Use SHM object cache to reduce memory ops

SHM Object can be reused to handle calls to OP-TEE system. Additional
objects are allocated as needed and stay in the cache. maximum number
entries in the cache is limited by CONFIG_OPTEE_SHM_CACHE_MAX_ENTRIES.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Signed-off-by: Mykyta Poturai <mykyta_poturai@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

oleksiimoisieiev committed:
tests: drivers: optee: Handle DISABLE_SHM_CACHE call in tests

optee driver send DISABLE_SHM_CACHE on init to make a resource cleaup
from the previous sessions. This call should be handled in tests to
avoid error message.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

xakep-amatop committed:
drivers: pinctrl: r8a779f0: move pfc file from soc to driver dir

Move pfc_r8a779f0.c from the 'soc' directory to the 'drivers'
directory to avoid code duplication for the ARM64 Spider board.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>

xakep-amatop committed:
drivers: clock_control: r8a779f0: add clocks related to MMC

Add the possibility to configure clocks related to MMC.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>

xakep-amatop committed:
soc: renesas_rcar: make common pinctrl header for both arm and arm64 archs

Make common pinctrl header for both arm and arm64 archs in order to avoid
code duplicates.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>

xakep-amatop committed:
drivers: pinctrl: pfc_rcar: add dummy function to pinctrl driver

Add a dummy pin function to the RCar PFC driver. It is necessary
to ensure that the driver sets the 'peripheral' bit (the driver
resets this bit during the first call of 'pfc_rcar_set_gpsr') for
a pin that doesn't have a pin function defined in IPSR, but always
acts as a 'peripheral', for example, the MMC pins on the Spider
board.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>

xakep-amatop committed:
snippets: xen_dom0: add support of rcar_spider_ca55

Add support of rcar_spider_ca55 to xen_dom0 snippet.

Disable CONFIG_UART_INTERRUPT_DRIVEN for xen_dom0, because
xen dom0 consoleio doesn't support interrupt driven mode.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>

xakep-amatop committed:
drivers: pinctrl: move renesas pincontrol drivers to separate directory

Moved r8a77951/r8a77961 pfc tables and funcs from soc to drivers/pfc dir.

Move all Renesas pinctrl driver source and header files to a separate
directory "renesas" and every SoC series to a separate subdirectory.

Update the MAINTAINERS file to align with the changes in the
pinctrl directory.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>

dsemenets committed:
drivers: optee: Remove warning message

Remove warning message. Optee test suite did not find any issues
with memrefs. Also we implemented memory registry in optee-client
which performs additional memref check.

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Acked-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

oleksiimoisieiev committed:
drivers: tee: optee: Use arch_page_phys_get for address conversion

According to the description of z_mem_phys_addr it uses
Z_MEM_VM_OFFSET to make a conversion from virtual to physical
address. And these macros are intended for assembly, linker code,
and static initializers.
So when external libraries, such as PICOLIBC or NEWLIB_LIBC
uses internal mappings for dynamic memory allocation z_mem_phys_addr
may not work properly and return wrong physical address.

We encountered this problem while testing the generation of a
certificate using PKCS11 in libckteec with MBEDTLS.
Mbedtls allocates buffer dynamically and ivokes tee command to request
certificate from PKCS11 TA which should place result to the allocated
buffer.
The result is the MBEDTLS is used with minimal libc - then everything
works fine. But when CONFIG_PICOLIBC or CONFIG_NEWLIB_LIBC is enabled
- it returns empty buffer. This happens because z_mem_phys_addr
returns incorrect physical address.

Signed-off-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

firscity committed:
Revert "ci: manifest: Use 'zephyrbot' token to run manifest action"

This reverts commit cb331139a879d8ed741d91f3f86ab89502798e46.
Mainline Zephyr repo uses separate permission for Github actions and
custom Zephyr bot (they have different tokens in env). This lead to
CI run failures, because custom actions token is not set. Since we do
not have Zephyrbot on our Github actions setup, so it is okay to use
regular Github token during action run.

Signed-off-by: Dmytro Firsov <dmytro_firsov@epam.com>
Acked-by: Oleksii Moisieiev <oleksii_moisieiev@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

Deedone committed:
fs: explicitly initialize fs_file_t and fs_dir_t

C++ compilers forbid usage of compound literals with -Wpedantic enabled.
They also forbid the usage of designated initializers for standards
lower than c++20.
So switch to explicit initialization, that would not make compilers
angry at us.

Signed-off-by: Mykyta Poturai <mykyta_poturai@epam.com>

firscity commented on snippets/xen_dom0/boards/qemu_cortex_a53.overlay:
```diff
@@ -16,13 +16,15 @@
 	 * following line in Xen boot log to set it right:
 	 * (XEN) Grant table range: 0x00000040200000-0x00000040240000
+	 * Also, add extended region 1:
+	 * (XEN) Extended region 1: 0x40000000->0x47e00000
```
`boards: shields: add extended region description to dom0 shield` -> `snippets: add extended region description to dom0 snippets`

firscity commented on modules/nanopb/CMakeLists.txt:
```diff
@@ -6,15 +6,25 @@ if(CONFIG_NANOPB)
 
   set(NANOPB_DIR ${ZEPHYR_CURRENT_MODULE_DIR})
 
+  find_program(PROTOC protoc)
```
Are these commits spurious? If no, please provide argumentation in commit message
`b18abbe Revert "modules: nanopb: Add custom target for generated header files"`
`61fdde05 Revert "modules: nanopb: Add helper function to generate sources"`

Deedone replied:
They break AOS protobuf generation, it's a workaround to fix build that I forgot about, will try to fix this properly

xakep-amatop commented:
It would be nice to wait until I remake 3ee924e5203a08c77225f49734f7cd388b0a8504 and use the new version instead. From my side, the other changes are okay.

-----
P.S. pls, fix checkpatch issues

xakep-amatop committed:
xen: align ram bank 0 configuration with libxl

Align GUEST_RAM0_SIZE with value in libxl.

Ensure that the first extended region does not overlap with
devices memory addresses.

Changes have been synced with the 'xen-4.19-xt0.1' branch
on the https://github.com/xen-troops/xen.git repository.

Note: this fix is not upstreamable. It is necessary to implement
      a proper fix while considering the memory ranges used in
      the DTS.

Signed-off-by: Mykola Kvach <mykola_kvach@epam.com>
Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>

Deedone commented:
Fixed spider board name for tests and samples

Deedone commented:
Replaced `fs: use memset to initialize fs_file_t and fs_dir_t` with the equivalent upstream patch.

GrygiriiS commented on snippets/xen_dom0/boards/qemu_cortex_a53.overlay:
```diff
@@ -16,13 +16,15 @@
 	hypervisor: hypervisor@40200000 {
 		compatible = "xen,xen";
-		reg = <0x0 0x40200000 0x0 0x40000>;
+		reg = <0x0 0x40200000 0x0 0x40000 0x0 0x40400000 0x0 0x17C00000>;
```
What was added here does not correspond to what is mentioned in comment above?

Deedone replied:
I am not the author of this change, @oleksiimoisieiev could you please take a look?
